app/grandchallenge/algorithms/serializers.py

```diff
@@ -84,13 +84,13 @@ class JobSerializer(serializers.ModelSerializer):
     """Serializer without hyperlinks for internal use"""
 
     algorithm_image = StringRelatedField()
+    algorithm = StringRelatedField()
```

Comment:
Is this field being filled out anywhere? It is used in predictions.json.

Reply:
BTW, it is unnecessary for the predictions.json, and could be used only in the serialiser for API view. Make sure the API list view does not have an n + 1 problem.

Reply:
It turns out that the API list view already HAD an n +1 problem. The culprit was the optional_hanging_protocols.

I am not sure what the best practice for testing the n + 1 problems is. I've added a somewhat naive test for it, but any feedback on that is welcome.

Reply:
Oh, and I can confirm the predictions.json is solely reliant on the precursor of the serializer: JobSerializer. So it doesn't show up at the predictions.json.